src/server/repositories/UrlRepository.ts

```diff
@@ -175,6 +175,27 @@ export class UrlRepository implements UrlRepositoryInterface {
     return this.urlMapper.persistenceToDto(newUrl)
   }
 
+  /**
+   * @param  {string} shortUrl Short url.
+   * @returns {Promise<boolean>} Returns true if shortUrl is available and false otherwise.
+   */
+  public isShortUrlAvailable: (shortUrl: string) => Promise<boolean> = async (
+    shortUrl,
+  ) => {
+    try {
+      // Cache lookup
+      // if long url does not exist, throws error
+      const longUrl = await this.getLongUrlFromCache(shortUrl)
+      return !longUrl
```

Comment:
nit: small optimisation but unnecessary
```suggestion
      return true
```